src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -3368,10 +3368,21 @@ def _binary_op_handler(self, node, parent):
             # Operator not supported, it will produce a CodeBlock instead
             raise NotImplementedError(operator_str)
 
+        # Capture the names of any named args
+        named_args = []
+        new_arg_nodes = []
+        for arg_node in arg_nodes:
+            if isinstance(arg_node, Fortran2003.Actual_Arg_Spec):
+                named_args.append(arg_node.children[0].string)
+                new_arg_nodes.append(arg_node.children[1])
+            else:
+                named_args.append(None)
+                new_arg_nodes.append(arg_node)
         binary_op = BinaryOperation(operator, parent=parent)
+        binary_op._named_args = named_args
```

Comment:
Updated all uses of `_named_args` with `named_args`

Reply:
I wrote a small test and fparser does not check for this at the moment.

Reply:
Your questions comes back to how language independent we are in the PSyIR. I don't see any reason for restricting the PSyIR in general so would suggest it should be a Fortran backend check. Would you be OK with this @sergisiso?

Reply:
The only problem is that the semantics must be clear to match a signature of a routine (are positional arguments after a named one shifted if the named one has to change its place?). Since we don't do call-callee checks yet I am not too worried at least in this PR if the PSyIR supports it and it is a backend check.

So the answer is yes, I will be OK with this for now.

Reply:
Added.